### PR TITLE
HTCONDOR-1459 capture stdout/err of transfer plugins and optionally write it in the StarterLog

### PR DESCRIPTION
**Pull request by johnkn**
Writing to the log is enabled/disabled by config.

<Insert PR description here, and leave checklist below for code review.>

# HTCondor Pull Request Checklist for internal reviewers

- [x] Verify that (GitHub thinks) the merge is clean. If it isn't, and you're confident you can resolve the conflicts, do so. Otherwise, send it back to the original developer.
- [x] Verify that the related Jira ticket exists and has a target version number and that it is correct.
- [x] Verify that the Jira ticket is in review status and is assigned to the reviewer.
- [x] Verify that the Jira ticket (HTCONDOR-xxx) is mentioned at the beginning of the title. Edit it, if not
- [x] Verify that the branch destination of the PR matches the target version of the ticket
- [x] Check for correctness of change
- [x] Check for regression test(s) of new features and bugfixes (if the feature doesn't require root)
- [x] Check for documentation, if needed  (documentation [build logs](https://readthedocs.org/projects/htcondor/builds/))
- [x] Check for version history, if needed
- [x] Check BaTLab dashboard for successful build (https://batlab.chtc.wisc.edu/results/workspace.php) and test for either the PR or a workspace build by the developer that has the Jira ticket as a comment.
- [x] Check that each commit message references the Jira ticket (HTCONDOR-xxx)

## After the above
- Hit the merge button if the pull request is approved and it is not a security patch (security changes require 2 additional reviews)
- If the pull request is approved, take the ticket out of review state
- Assign JIRA Ticket back to the developer

### CODE REVIEW

File: src/condor_utils/file_transfer.cpp [hunk 1]
```diff
@@ -6576,44 +6574,44 @@ FileTransfer::InvokeMultipleFileTransferPlugin( CondorError &e,
...
-	// Invoke the plugin
-	dprintf( D_ALWAYS, "FILETRANSFER: invoking: %s \n", plugin_path.c_str() );
-	dprintf( D_FULLDEBUG, "FILETRANSFER: INPUT FILE: %s\n", transfer_files_string.c_str() );
-	FILE* plugin_pipe = my_popen( plugin_args, "r", FALSE, &plugin_env, drop_privs );
-	if( !plugin_pipe ) {
-		dprintf ( D_ALWAYS, "FILETRANSFER: failed to invoke multifile transfer "
-			"plugin %s, aborting\n", plugin_path.c_str() );
-		return TransferPluginResult::Error;
-	}
+	bool want_stderr = param_boolean("REDIRECT_FILETRANSFER_PLUGIN_STDERR_TO_STDOUT", true);
```
Comment (Todd-L-Miller) [inline src/condor_utils/file_transfer.cpp:6577]: Strictly speaking, we should also be doing this for single-file plug-ins.

File: src/condor_utils/file_transfer.cpp [hunk 2]
```diff
@@ -6576,44 +6574,44 @@ FileTransfer::InvokeMultipleFileTransferPlugin( CondorError &e,
 		plugin_args.AppendArg( "-upload" );
 	}
 
-	// Invoke the plugin
-	dprintf( D_ALWAYS, "FILETRANSFER: invoking: %s \n", plugin_path.c_str() );
```
Comment (Todd-L-Miller) [inline src/condor_utils/file_transfer.cpp:6580]: Removing this line breaks test_curl_plugin.  I have no preference between adding this line back in (as D_TEST?) or changing the test.

**Review by Todd-L-Miller — COMMENTED** [review]
I'm not too worried about logging stderr vs stdout being unimplemented for single-file plug-ins, but the tests need to pass.

**Comment by Todd-L-Miller** [comment]
Also seems like it shouldn't be hard to write a test that turns on the appropriate knobs for a do-nothing plug-in that just dumps known text into the log.

**Comment by timtheisen** [comment]
Is this one of the fixes that we want to sneak into HTCondor 23.0.3?

**Comment by bbockelm** [comment]
Notes:
- New parameters are not listed in the param table.
- No documentation on how to turn this on/off.

File: src/condor_utils/file_transfer.cpp (hunk 1 above)
Comment (johnkn) [reply src/condor_utils/file_transfer.cpp:6577]: stdout of single file plugins is defined to be the status ad.

**Comment by Todd-L-Miller** [comment]
I've managed to forget: are
* `REDIRECT_FILETRANSFER_PLUGIN_STDERR_TO_STDOUT`,
* `LOG_FILETRANSFER_PLUGIN_STDOUT_ON_SUCCESS`, and
* `LOG_FILETRANSFER_PLUGIN_STDOUT_ON_FAILURE`
deliberately undocumented?

**Comment by johnkn** [comment]
> I've managed to forget: are
> 
> * `REDIRECT_FILETRANSFER_PLUGIN_STDERR_TO_STDOUT`,
> * `LOG_FILETRANSFER_PLUGIN_STDOUT_ON_SUCCESS`, and
> * `LOG_FILETRANSFER_PLUGIN_STDOUT_ON_FAILURE`
>   deliberately undocumented and/or not in the param table?

Deliberately not documented and not in the param table,  For this feature to be useful, we may need the behavior controlled by these knobs using other mechanisms.  Thus documenting the knobs as a supported is not a good idea at this time.

**Comment by Todd-L-Miller** [comment]
That makes sense.  Thanks for the explanation.

**Review by Todd-L-Miller — APPROVED** [review]
Always good to remove code.

**Pull request merged**